source/validate_types.cpp

```diff
@@ -185,23 +189,127 @@ ModuleLayoutSection ValidationState_t::getLayoutStage() const {
 }
 
 void ValidationState_t::progressToNextLayoutStageOrder() {
-  module_layout_order_stage_ +=
-      module_layout_order_stage_ < GetModuleOrder().size();
-  if (module_layout_order_stage_ >= GetModuleOrder().size()) {
-    current_layout_stage_ = libspirv::ModuleLayoutSection::kFunction;
+  if (current_layout_stage_ <= GetModuleOrder().size()) {
+    current_layout_stage_ = (ModuleLayoutSection)(current_layout_stage_ + 1);
   }
 }
 
 bool ValidationState_t::isOpcodeInCurrentLayoutStage(SpvOp op) {
-  const vector<SpvOp>& currentStage =
-      GetModuleOrder()[module_layout_order_stage_];
+  const vector<SpvOp>& currentStage = GetModuleOrder()[current_layout_stage_];
   return end(currentStage) != find(begin(currentStage), end(currentStage), op);
```

Comment:
[Koenig lookup](https://en.wikipedia.org/wiki/Argument-dependent_name_lookup)

since the vector is part of the `std` namespace, the appropriate `std::begin` and `std::end` functions are called.